src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -709,8 +709,24 @@ FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now)
 
 	_skipping_takeoff_detection = false;
 
-	if (((_control_mode.flag_control_auto_enabled && _control_mode.flag_control_position_enabled) ||
-	     _control_mode.flag_control_offboard_enabled) && _position_setpoint_current_valid) {
+	if (_control_mode.flag_control_offboard_enabled && _position_setpoint_current_valid) {
+		if (_control_mode.flag_control_position_enabled) {
+			if (PX4_ISFINITE(_pos_sp_triplet.current.vx) && PX4_ISFINITE(_pos_sp_triplet.current.vy)
+			    && PX4_ISFINITE(_pos_sp_triplet.current.vz)) {
+				// Offboard position with velocity setpoints
+				_control_mode_current = FW_POSCTRL_MODE_AUTO_PATH;
+				return;
+
+			} else {
+				// Offboard position setpoint only
+				_control_mode_current = FW_POSCTRL_MODE_AUTO;
+				return;
+			}
+		}
+	}
+
+	if ((_control_mode.flag_control_auto_enabled && _control_mode.flag_control_position_enabled)
```

Comment:
Follow up to the question above: can we make this into an else if() to only handle non-offboard cases?

Reply:
Done!